engine/apps/google/tasks.py

```diff
@@ -15,9 +16,69 @@ def sync_out_of_office_calendar_events_for_user(google_oauth2_user_pk: int) -> N
     google_oauth2_user = GoogleOAuth2User.objects.get(pk=google_oauth2_user_pk)
     google_api_client = GoogleCalendarAPIClient(google_oauth2_user.access_token, google_oauth2_user.refresh_token)
 
-    # NOTE: shift swap request generation will be done in https://github.com/grafana/oncall-private/issues/2590
-    # QUESTION: will we need to persist any information about these calendar events in our database?
-    google_api_client.fetch_out_of_office_events()
+    user = google_oauth2_user.user
+    user_id = user.pk
+
+    logger.info(f"Syncing out of office Google Calendar events for user {user_id}")
+
+    user_google_calendar_settings = user.google_calendar_settings
+    specific_oncall_schedules_to_sync = user_google_calendar_settings["specific_oncall_schedules_to_sync"]
+
+    users_schedules = OnCallSchedule.objects.related_to_user(user)
+    if specific_oncall_schedules_to_sync:
+        users_schedules = users_schedules.filter(public_primary_key__in=specific_oncall_schedules_to_sync)
```

Comment:
I see the usage of the `specific_oncall_schedules_to_sync` now. At first read I thought it would sync those calendars with on-call shifts, but now I see the idea is to list the schedules for which you want to automatically request the shifts. Wondering if this could be renamed, or at least add some comment/note in the field definition to make it clearer what means to have an schedule ID listed there?

Reply:
that's a good point. After looking at this a second time, I agree, the naming isn't optimal. Maybe something like `oncall_schedules_to_consider_for_shift_swaps`? (a bit long, but more clear of its intention).

I'll also update the UI to be more descriptive on this field 👍